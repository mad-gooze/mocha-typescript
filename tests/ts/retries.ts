import * as mocha from "mocha";
import { suite, test, context } from "../../index";

@suite('retries') class RetriesTest {
    // Get the mocha context in for instance before and after (before/after each) and test methods.
    @context mocha: mocha.IBeforeAndAfterContext & mocha.IHookCallbackContext;

    // Get the mocha context for static before and after.
    @context static mocha: mocha.IBeforeAndAfterContext & mocha.IHookCallbackContext;

    static before() {
        this.mocha.retries(1);   
    }

    @test "retried once"() {
        console.log("test");
        throw new Error("should fail");
    }
}
