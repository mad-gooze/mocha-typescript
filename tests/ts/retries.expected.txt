

  retries
  retries
test
    1) retried once


  0 passing (11ms)
  1 failing

  1) retries retried once:
     Error: should fail
      at RetriesTest.retried once (tests/ts/retries.js:16:15)
      at Context.<anonymous> (index.js:173:43)



